tasks/main.yml

```diff
@@ -89,7 +89,7 @@
   when: env == 'production' or env == 'staging'
 
 - include: ssl_certificate.yml
-  when: laravel_https_enabled
+  when: laravel_https_enabled and (env == 'production' or env == 'staging')
```

Comment:
No need to check environment here. The variable can be set for different environments in playbook inventory files.